### ISSUE TITLE
Custom integraton with injection breaks request

### ISSUE THREAD

**Issue by Growiel** [getsentry/sentry-symfony#575]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
PHP (Symfony) 4.1.2

### Steps to Reproduce

1. Create a custom Integration with a service injected in it. (contructor injection and setter injection both do the same thing)

PHP file:
```
<?php

declare(strict_types=1);

namespace App\Sentry\Integration;

use App\Service;
use Sentry\Event;
use Sentry\Integration\IntegrationInterface;
use Sentry\SentrySdk;
use Sentry\State\Scope;

class AddContextIntegration implements IntegrationInterface
{
    private Service $service;

    /**
     * @required
     */
    public function setService(Service $service): void
    {
        $this->service= $service;
    }

    public function setupOnce(): void
    {
        Scope::addGlobalEventProcessor(static function (Event $event): ?Event {
            $integration = SentrySdk::getCurrentHub()->getIntegration(self::class);
            
            if (null !== $integration) {
                $integration->addContext($event);
            }

            return $event;
        });
    }

    private function addContext(Event $event): Event
    {
        $event->setContext('plop', ['haha' => 'super test']);

        return $event;
    }
}
```

Relevant part of the YAML config:

```
sentry:
  options:
    integrations:
      - Sentry\Integration\IgnoreErrorsIntegration
      - App\Sentry\Integration\AddContextIntegration
```

### Expected Result

I expected my integration to work and to correctly add context before sending my events.

### Actual Result

As soon as I add the line in the YAML, my `RequestStack` is empty:

![image](https://user-images.githubusercontent.com/5228735/143056026-f7a1c881-06f8-444a-884c-1e7167ded3ef.png)

If I remove the line in YAML, and only this one, I correctly have this:

![image](https://user-images.githubusercontent.com/5228735/143056162-9d8b914a-7a6a-41dc-af5b-b396d814ad76.png)

I'm not sure what happens when Sentry registers a new Integration, but something removes the request from the RequestStack.

I tried using my integration without Sentry (inject it into a Controller and call `setupOnce()` manually) without any issues, the context was correctly added to my events.

This is the workaround: create an early request listener and manually call my Integration, but i'd like to use the official way.

Thanks.

**Comment by ste93cry** [comment]
Weird, we never call the `RequestStack::pop()` method. Can you reproduce the behaviour also on a newly created project?

**Comment by Growiel** [comment]
Hi, thanks for answering.

No bug on your side, the RequestStack is indeed empty because the service is initialized BEFORE the `requestEvent` is dispatched (because of the Monolog Handler), but the real issue was that in your service we did this:

```
function __consutrct(RequestStack $requestStack){
    $this->request = $requestStack->getCurrentRequest();
}
```

We saved a request equal to null, and that caused our issue. We changed it to match what the documentation says (keep the RequestStack as a propery and call `getCurrentRequest()` when needed) and it now works.

No issues with Sentry.

Sorry for the wasted time.
